hummingbird/ml/operator_converters/_tree_commons.py

```diff
@@ -180,22 +180,11 @@ def get_parameters_for_sklearn_common(tree_infos):
     trees = tree_infos
     if hasattr(trees, "nodes"):
         # SklearnHistGradientBoostingClassifier
-
-        lefts = [trees.nodes[x]["left"] for x in range(len(trees.nodes))]
-        # Use -1 instead of 0 for nodes with no lefts
-        lefts = [idx if idx != 0 else -1 for idx in lefts]
-
-        rights = [trees.nodes[x]["right"] for x in range(len(trees.nodes))]
-        # Use -1 instead of 0 for nodes with no rights
-        rights = [idx if idx != 0 else -1 for idx in rights]
-
-        features = [trees.nodes[x]["feature_idx"] for x in range(len(trees.nodes))]
-        thresholds = [trees.nodes[x]["threshold"] for x in range(len(trees.nodes))]
-
-        values = [trees.nodes[x]["value"] for x in range(len(trees.nodes))]
-        # Convert to numpy.ndarray with shape (s, 1, 1)
-        values = np.array(values)
-        values = values.reshape((values.shape[0], 1, 1))
+        features = [n[2] if n[2]!=0 else 0 for n in trees.nodes]
+        thresholds = [n[3] if n[3]!=0 else -1 for n in trees.nodes]
+        lefts = [n[4] if n[4]!=0 else -1 for n in trees.nodes]
+        rights = [n[5] if n[5]!=0 else -1 for n in trees.nodes]
+        values = [[n[0]] if n[0]!=0 else [-1] for n in trees.nodes]
```

Comment:
The tests are giving errors this time instead of failures. The reason is the `_find_max_depth()` in `_tree_commons.py`.
When I reverted these lines back to using `lefts = [trees.nodes[x]["left"] for x in range(len(trees.nodes))]` ... etc, all 10 tests passed.
I will inspect this now.

Reply:
@interesaaat I found the issue!
The problem with accessing tree nodes by index is that it's not guaranteed that this index will always be equivalent to the property that you're trying to fetch.
Using the string name of the property for fetching is before guarantees the correct result.